lib/util/propTypes.js

```diff
@@ -368,10 +368,15 @@ module.exports = function propTypesInstructions(context, components, utils) {
       node
       && node.type === 'Identifier'
     ) {
-      const scope = context.getScope();
-      const identVariable = scope.variableScope.variables.find(
-        (variable) => variable.name === node.name
-      );
+      const scope = astUtil.getScope(context, node);
+      const allVars = [];
+      let vars = (scope.childScopes[0] || scope).variableScope;
```

Comment:
can you elaborate on the need for this? did eslint 9 just straight up remove this functionality?